src/psyclone/tests/domain/lfric/kernel/lfric_kernel_metadata_test.py

```diff
@@ -678,3 +1404,43 @@ def test_setter_getter_name():
             "'1_invalid'." in str(info.value))
     metadata.name = "kern_type"
     assert metadata.name == "kern_type"
+
+
+def test_meta_args_get():
+    '''Test that the meta_args_get method behaves as expected.'''
+    # Empty meta_args list.
+    metadata = LFRicKernelMetadata()
+    assert metadata.meta_args_get([]) == []
+
+    # types is not a list or CommonMetaArgMetadata subclass.
+    with pytest.raises(TypeError) as info:
+        metadata.meta_args_get(None)
+    assert ("Expected a subclass of CommonMetaArgMetadata or a list for the "
+            "types argument, but found 'NoneType'." in str(info.value))
+
+    # types has invalid values in its list.
+    with pytest.raises(TypeError) as info:
+        metadata.meta_args_get([None])
+    assert ("Expected list entries in the types argument to be subclasses of "
+            "CommonMetaArgMetadata, but found 'NoneType'." in str(info.value))
+
+    meta_args = [
+        ScalarArgMetadata("gh_real", "gh_read"),
+        FieldArgMetadata("gh_real", "gh_read", "w0"),
+        OperatorArgMetadata("gh_real", "gh_read", "w0", "w1")]
+    metadata = LFRicKernelMetadata(meta_args=meta_args)
+
+    # OK (types single value)
+    result = metadata.meta_args_get(ScalarArgMetadata)
+    assert len(result) == 1
+    assert result[0] == metadata.meta_args[0]
```

Comment:
Possibly `is` rather than `==` and below?

Reply:
Modified, thanks.